lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -411,23 +443,29 @@ export class UserAgentApplication {
                         this.logger.error("Error occurred during unified cache ATS: " + error);
 
                         // proceed to login since ATS failed
-                        this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject);
+                        this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject, state);
                     });
                 }
                 // No ADAL token found, proceed to login
                 else {
-                    this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject);
+                    this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject, state);
                 }
             }
             // AcquireToken call, but no account or context given, so throw error
             else {
                 this.logger.info("User login is required");
-                throw ClientAuthError.createUserLoginRequiredError();
+                const stateOnlyResponse = buildResponseStateOnly(this.getAccountState(state));
+                this.cacheStorage.resetTempCacheItems(state);
+                this.authErrorHandler(interactionType,
+                    ClientAuthError.createUserLoginRequiredError(),
+                    stateOnlyResponse,
+                    reject);
+                return;
```

Comment:
Is this `return` needed?

Reply:
All the callback code in UAA, `authErrorHandler` or `authResponseHandler` are followed by `return` statements, mostly to ensure the function execution is halted I guess. I retained the pattern. We may get rid of them, but I want to test before we do that and let us take it as a separate enhancement.